### PR TITLE
Fix issues I ran into when porting our code to the new API

### PR DESCRIPTION
**Pull request by hrydgard**
First round of changes. Will require a major API version bump once it's all in.

Generally this wasn't too bad but there are holes.

A lot of the SceneDesc parameters were lost, added back a few. It's gonna be more though, so should we add a builder?

Now includes @Hentropy 's new object creation rework.

### CODE REVIEW

File: physx/src/physics.rs [hunk 1]
```diff
@@ -100,9 +100,9 @@ pub const PX_PHYSICS_VERSION: u32 = crate::version(4, 1, 1);
 /// Parametrized by the Foundation's Allocator and the Physics' Shape type.
 pub struct PhysicsFoundation<Allocator: AllocatorCallback, Geom: Shape> {
     // Order matters here for Drop. Foundation must be dropped last.
-    pub physics: Owner<PxPhysics<Geom>>,
+    physics: Owner<PxPhysics<Geom>>,
     pub pvd: Option<VisualDebugger>,
```
Comment (tgolsson) [inline physx/src/physics.rs:104]: Let's be consistent and make this one private too while we're at it.

**Comment by hrydgard** [comment]
PxSceneDesc::new is really getting too many arguments, as mentioned. Won't scale. Builder or parameter struct?

**Comment by tgolsson** [comment]
I'm a fan of the builder pattern but in this case I think a parameter struct would be better. Maybe just convert PxSceneDesc to that?

**Comment by Hentropy** [comment]
I was thinking of a descriptor pattern for the objects created by physics.  Something like the following:
```rust
pub trait Physics {
    fn create<Desc: Descriptor>(&mut self, desc: Desc) -> Desc::Target {
        desc.create(self)
    }
// ... rest of Physics interface
}

pub trait Descriptor {
    type Target;
    fn create(&self, physics: &mut impl Physics) -> Self::Target;
}
```

I have a pretty clear idea of what to do here, I just never got around to implementing it. I'll write it up and add it to this PR.

**Comment by Hentropy** [comment]
I can't add directly to this so see #114

File: physx/src/scene.rs [hunk 2]
```diff
@@ -923,31 +968,89 @@ impl From<PxPairFilteringMode::Enum> for PairFilteringMode {
...
+    }
+}
+
 #[derive(Debug, Clone, Copy)]
 #[repr(u32)]
 pub enum SolverType {
-    PGS = 1,
-    TGS = 2,
+    PGS = 0,
+    TGS = 1,
```
Comment (tgolsson) [inline physx/src/scene.rs:1053]: Can't we use the Px constants here directly?

**Review by tgolsson — CHANGES_REQUESTED** [review]

**Comment by tgolsson** [comment]
@hrydgard  Also please update the changelog!

**Comment by Hentropy** [comment]
#117 could be rolled into this as well, I left it marked as a draft so I wasn't pinging Maik and Tomasz.

**Comment by hrydgard** [comment]
Yeah, I'll roll that in later today.

File: physx/src/scene.rs (hunk 2 above)
Comment (hrydgard) [reply physx/src/scene.rs:1053]: I guess we could, was just following the pattern of other enums. Don't think it really matters though.

**Comment by hrydgard** [comment]
I think we can get this in as-is, and do any further fixes separately before we tag a new release.

**Review by tgolsson — APPROVED** [review]
LGTM

**Pull request merged**